Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -626,29 +641,22 @@ void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
 }
 
 /**
- @Status Interoperable
- @Notes
+ @Status Caveat
+ @Notes eoFill ignored. Default fill pattern for ID2D1 Geometry is used.
 */
 bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* transform, CGPoint point, bool eoFill) {
+    RETURN_FALSE_IF(!path);
+
     if (transform) {
         point = CGPointApplyAffineTransform(point, *transform);
     }
-    // check if the point is outside this box already, if it is, return false
-    CGRect boundingBox = CGPathGetBoundingBox(path);
-    if (!CGRectContainsPoint(boundingBox, point)) {
-        return false;
-    }
 
-    CGContextRef context =
-        CGBitmapContextCreate(0, boundingBox.origin.x + boundingBox.size.width, boundingBox.origin.y + boundingBox.size.height, 1, 1, 0, 0);
+    BOOL containsPoint = FALSE;
 
-    CGContextAddPath(context, path);
-
-    bool inPath = CGContextIsPointInPath(context, eoFill, point.x, point.y);
-
-    CGContextRelease(context);
+    path->closePath();
+    FAIL_FAST_IF_FAILED(path->_impl.pathGeometry->FillContainsPoint(_CGPointToD2D_F(point), D2D1::IdentityMatrix(), &containsPoint));
```

Comment:
The intention I used with fail_fast was that if the geometry APIs ever resulted in an unrecoverable state for the path that this should fail. This one in particular should have been to return false instead of failing.

---
In reply to: [85594909](https://github.com/Microsoft/WinObjC/pull/1255#discussion_r85594909) [](ancestors = 85594909)